### PR TITLE
Add palindrome checker implementations with test coverage in C++, Python, and Java

### PR DESCRIPTION
**Pull request by SWASTIKpy**
## What?
Implemented palindrome string checker in three languages (C++, Python, Java) within a single file, with comprehensive test coverage and optional features for case insensitivity and space handling.

## Why?
Fulfills issue requirements for adding palindrome validation to the strings folder with efficient two-pointer implementation across multiple languages, following project convention of multi-language examples in single files.

## Implementation Details
- **Algorithm**: Two-pointer approach (left & right pointers converging)
- **Time Complexity**: O(n) - single pass through string
- **Space Complexity**: O(1) - only two pointer variables
- **Languages**: C++, Python (commented), Java (commented) with consistent, well-commented logic across all implementations
- **Optional Features**: Case-insensitive comparison and space-ignoring capability

## Test Coverage
Included 9 comprehensive test cases:
- Basic palindromes: "racecar", "Madam"
- Palindromes with spaces: "Ra ce car", "A man a plan a canal Panama", "No lemon, no melon"
- Non-palindromes: "hello"
- Edge cases: empty string, single space, spaces around single character

## File Location
- `/strings/palindrome.cpp` (contains C++, Python, and Java implementations)

## Testing
All test cases produce expected boolean outputs (true/false) and handle edge cases correctly across all three language implementations.

## Summary by CodeRabbit

* **New Features**
  * Added palindrome-checking utilities in C++, Java, and Python that ignore whitespace and are case-insensitive, each with a built-in test harness that runs multiple sample inputs and prints true/false.
  * Consolidated the C++ implementation into a clearer, test-driven flow.

* **Style**
  * Minor comment/formatting update in the reverse-string implementation; no behavior changes.

### CODE REVIEW

File: strings/palindrome.cpp [hunk 1]
```diff
@@ -1,26 +1,49 @@
...
 
-using namespace std;
+    while (left < right) {
+        // Skip spaces from the left
+        while (left < right && isspace(str[left])) left++;
+        // Skip spaces from the right
+        while (left < right && isspace(str[right])) right--;
+
+        // Case insensitivity check
+        if (tolower(str[left]) != tolower(str[right]))
```
[the remark this answers is not in the document]
Comment (SWASTIKpy) [reply strings/palindrome.cpp:21]: people here are trying to learn dsa it will make things more complex for them to understand i can do it as well if maintainer asks
